src/plugins/intel_gpu/src/graph/impls/common/condition.cpp

```diff
@@ -139,11 +140,12 @@ struct condition_impl : typed_primitive_impl<condition> {
         // Set output memory of condition_inst to inner network output memory after inner network execution
         instance.postprocess_output_memory(executed_net, branch);
 
-        for (auto& output : sub_net_results)
-            if (output.second.get_event() != nullptr)
-                output_events.push_back(output.second.get_event());
+        std::vector<event::ptr> executed_events;
+        for (const auto& ev : executed_net->get_executed_primitives()) {
```

Comment:
I guess processing of all internal events may impact the performance w/o profiling. So I suggest changing it to something like this:
```
if (enable_profilng) {
// process all events
} else {
// process out events only
}
```

@yeonbok What do you think?